core/src/main/java/org/eclipse/hono/config/KeyLoader.java

```diff
@@ -90,79 +129,106 @@ public static KeyLoader fromKeyStore(final Vertx vertx, final String keyStorePat
      * @throws IllegalArgumentException if any of the files does not exist.
      */
     public static KeyLoader fromFiles(final Vertx vertx, final String keyPath, final String certPath) {
-        KeyLoader result = new KeyLoader(vertx);
-        result.loadKeysFromFiles(keyPath, certPath);
-        return result;
-    }
 
-    public PrivateKey getPrivateKey() {
-        return privateKey;
+        PrivateKey privateKey = null;
+        PublicKey publicKey = null;
+
+        if (keyPath != null) {
+            privateKey = loadPrivateKeyFromFile(vertx, keyPath);
+        }
+        if (certPath != null) {
+            publicKey = loadPublicKeyFromFile(vertx, certPath);
+        }
+
+        return new KeyLoader(privateKey, publicKey);
+
     }
 
-    public PublicKey getPublicKey() {
-        return publicKey;
+    private static PrivateKey generateRsaKey(KeySpec keySpec) throws GeneralSecurityException {
+        return KeyFactory
+                .getInstance("RSA")
+                .generatePrivate(keySpec);
     }
 
-    private void loadKeysFromFiles(final String keyPath, final String certPath) {
+    private static <R> R processFile(final Vertx vertx, final String pathName, final PemProcessor<R> processor) {
 
-        if (keyPath != null) {
-            loadPrivateKeyFromFile(keyPath);
+        final Path path = Paths.get(pathName);
+
+        if (!vertx.fileSystem().existsBlocking(pathName)) {
+            throw new IllegalArgumentException(format("%s: Private key file does not exist", path));
```

Comment:
I think the file does not necessarily contain a *private* key, does it? We use the same method to load public keys as well, don't we?